interp/constrextern.ml

```diff
@@ -307,8 +307,20 @@ let extern_evar n l = CEvar (n,l)
     For instance, in the debugger the tables of global references
     may be inaccurate *)
 
+let rec dirpath_of_modpath = function
+  | MPfile dp -> dp
+  | MPbound mbid -> let (_,id,_) = MBId.repr mbid in DirPath.make [id]
+  | MPdot (t, l) -> Libnames.add_dirpath_suffix (dirpath_of_modpath t) (Label.to_id l)
+
+let path_of_global = function
+  | GlobRef.VarRef id -> Libnames.make_path DirPath.empty id
+  | GlobRef.ConstRef cst -> Libnames.make_path (dirpath_of_modpath (Constant.modpath cst)) (Label.to_id (Constant.label cst))
```

Comment:
By uniformity and to signal an issue with the environment, you should also wrap the label of the constant inside chevrons as you did below.

Reply:
In this case, the name will be right. So, I find it delicate to tell suggest that the name is maybe wrong while it is actually ok. Do we really want to expose more of the internals to the user?

It depends also on how temporary is intended to be the PR, i.e. about how we plan to address the problem in the future.

- Is the plan e.g. to make the nametab a functional parameter of the printer? If yes, we had a few PR/ceps about it and maybe it is the right time to consider them again.
- Another safe and relatively easy-to-implement fix could be to hide the names of inductive type and constructors directly in a `MutInd.t` object. Since the names bound in a module parameter have to be qualified anyway (assuming `Import`s are dropped in printing), this would not make a big difference wrt not using the shortest qualid.

Reply:
> Do we really want to expose more of the internals to the user?

OK, fine, it's better as it is currently then. But it's hiding a real bug in functor printing.

> Is the plan e.g. to make the nametab a functional parameter of the printer?

I think so, functor printing is very broken. Worse, it's not only printing, we also have a lot of issues with functors and libojects. A principled solution should be found.

Reply:
> We also have a lot of issues with functors and libojects.

Right.